src/Engine/DatabaseEngine.php

```diff
@@ -168,6 +171,31 @@ private function queryFloat(string $function, Geometry|string|float|int ...$para
         return (float) $result;
     }
 
+    /**
+     * Queries a GIS function returning a MultiPoint value.
+     *
+     * @param string                       $function   The SQL GIS function to execute.
+     * @param Geometry|string|float|int ...$parameters The Geometry objects or scalar values to pass as parameters.
+     *
+     * @throws GeometryEngineException
+     */
+    private function queryMultiPoint(string $function, Geometry|string|float|int ...$parameters) : MultiPoint
+    {
+        $result = $this->queryGeometry($function, ...$parameters);
+
+        if ($result instanceof MultiPoint) {
+            return $result;
+        }
+
+        $geometry = $result instanceof ProxyInterface ? $result->getGeometry() : $result;
+
+        if ($geometry->isEmpty()) {
+            return MultiPoint::fromText('MULTIPOINT EMPTY');
```

Comment:
```suggestion
            return new MultiPoint($geometry->coordinateSystem());
```

This is to keep the same coordinate system as the original geometry (currently you're unconditionally creating an `XY` geometry with SRID `0`).

Reply:
great input - thanks!